pkg/rngd/cmd/rngd/rng_linux_arm64.go

```diff
@@ -7,15 +7,47 @@ package main
 import "C"
 
 import (
-	"errors"
+	"encoding/binary"
+	"log"
+	"os"
 )
 
-// No standard RNG on arm64
-
+// Check if there is a hardware RNG device on AArch64, will use that
+// device as the RNG entropy source if we do have
 func initRand() bool {
-	return false
+
+	f, err := os.Open("/dev/hwrng")
+	defer f.Close()
+	if err != nil {
+		log.Fatalf("Cannot open /dev/hwrng: %v", err)
+		return false
+	}
+
+	// Try to read some data from the entropy source. If it doesn't
+	// return an error, assume it's ok to use
+	buf := make([]byte, 16)
+	n, err := f.Read(buf)
```

Comment:
Is this a read on `/dev/hwrng` blocking if there is no entropy, like `/dev/random`? If so we want a timeout on this,

Reply:
I am not pretty sure since I can't test all the hardware RNG devices, but if that case happens, probably it's hardware bug.